src/model-helpers/Profile.js

```diff
@@ -0,0 +1,141 @@
+
+  const profileInfo = { profileName, nickname, profileImage };
+
+  // TODO: include more fields to the object like organization, address, etc.
+  // when expanding this feature
+  return { profileInfo, profileDataset, profileThing };
+};
+
+/**
+ * A function that updates the user's profile information based on what is inputed
+ * in the Profile page
+ *
+ * @function updateProfileInfo
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} profileData - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ * @param {object} inputValues - The inputs for updating profile information
+ * on their profile card
+ * @returns {Promise} Promise - Performs action to update profile card on the
+ * user's profile card
+ */
+export const updateProfileInfo = async (session, profileData, inputValues) => {
+  let { profileDataset, profileThing } = profileData;
+  const { profileInfo } = profileData;
+
+  Object.keys(inputValues).forEach((input) => {
+    switch (inputValues[input]) {
+      case null:
+        return;
+      case '':
+        profileThing = removeStringNoLocale(
+          profileThing,
+          RDF_PREDICATES[input],
+          profileInfo[input]
+        );
+        break;
+      default:
+        profileThing = buildThing(profileThing)
+          .setStringNoLocale(RDF_PREDICATES[input], inputValues[input])
+          .build();
+        break;
+    }
+  });
+
+  profileDataset = setThing(profileDataset, profileThing);
+
+  await saveSolidDatasetAt(session.info.webId, profileDataset, { fetch: session.fetch });
+};
+
+/**
+ * Function that upload profile image of user onto their profile container and
+ * reference it on their Pod
+ *
+ * @function uploadProfileImage
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} profileData - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ * @param {Blob} inputImage - File blob being uploaded for profile image
+ * @returns {Promise} Promise - Performs a file upload of your profile image onto
+ * your pod at /profile
+ */
+export const uploadProfileImage = async (session, profileData, inputImage) => {
+  let { profileDataset, profileThing } = profileData;
+
+  const profileContainer = `${session.info.webId.split('card')[0]}`;
+  const profileImgFilename = inputImage.name.replace(' ', '%20');
+
+  await saveFileInContainer(profileContainer, inputImage, {
+    slug: profileImgFilename,
+    contentType: inputImage.type,
+    fetch: session.fetch
+  });
+
+  profileThing = buildThing(profileThing)
+    .addUrl(RDF_PREDICATES.profileImg, `${profileContainer}${profileImgFilename}`)
+    .build();
+
+  profileDataset = setThing(profileDataset, profileThing);
+
+  await saveSolidDatasetAt(session.info.webId, profileDataset, { fetch: session.fetch });
```

Comment:
Probably want to add an acl to this so the profile pic can be read by the public.

Reply:
Got it included with `uploadProfileImage`. I've updated the unit test for it as well to take the new functions to build the ACL into account.